packages/relic/lib/src/headers/typed/headers/cookie_header.dart

```diff
@@ -52,71 +60,348 @@ class Cookie {
   /// The value of the cookie.
   final String value;
 
-  /// Constructs a [Cookie] instance with the specified name and value.
   Cookie({
-    required this.name,
-    required this.value,
-  }) {
-    if (!isNameValid(name)) {
-      throw FormatException('Invalid cookie name');
-    }
-    if (!isValueValid(value)) {
-      throw FormatException('Invalid cookie value');
-    }
-  }
+    required String name,
+    required String value,
+  })  : name = _validateName(name),
+        value = _validateValue(value);
 
   factory Cookie.parse(String value) {
-    final parts = value.split('=');
-    if (parts.length != 2) {
+    var splitValue = value.split('=');
+    if (splitValue.length != 2) {
       throw FormatException('Invalid cookie format');
     }
 
-    final name = parts[0].trim();
-    final rawValue = parts[1].trim();
+    return Cookie(
+      name: splitValue.first.trim(),
+      value: splitValue.last.trim(),
+    );
+  }
+
+  /// Converts the [Cookie] instance into a string representation suitable for HTTP headers.
+  String toHeaderString() => '$name=$value';
+
+  @override
+  String toString() {
+    return 'Cookie(name: $name, value: $value)';
+  }
+}
+
+/// A class representing the HTTP Set-Cookie header.
+///
+/// This class manages the parsing and representation of set cookie.
+class SetCookieHeader extends TypedHeader {
+  /// The keys used for the Set-Cookie header.
+  static const String _expires = 'Expires=';
+  static const String _maxAge = 'Max-Age=';
+  static const String _domain = 'Domain=';
+  static const String _path = 'Path=';
+  static const String _secure = 'Secure';
+  static const String _httpOnly = 'HttpOnly';
+  static const String _sameSite = 'SameSite=';
+
+  /// The name of the cookie.
+  final String name;
+
+  /// The value of the cookie.
+  final String value;
+
+  /// The time at which the cookie expires.
+  final DateTime? expires;
+
+  /// The number of seconds until the cookie expires. A zero or negative value
+  /// means the cookie has expired.
+  final int? maxAge;
+
+  /// The domain that the cookie applies to.
+  final Uri? domain;
+
+  /// The path within the [domain] that the cookie applies to.
+  final Uri? path;
+
+  /// Whether to only send this cookie on secure connections.
+  final bool secure;
 
-    if (!isNameValid(name)) {
-      throw FormatException('Invalid cookie name');
+  /// Whether the cookie is only sent in the HTTP request and is not made
+  /// available to client side scripts.
+  final bool httpOnly;
+
+  /// Whether the cookie is available from other sites.
+  ///
+  /// This value is `null` if the SameSite attribute is not present.
+  ///
+  /// See [SameSite] for more information.
+  SameSite? sameSite;
+
+  /// Constructs a [Cookie] instance with the specified name and value.
+  SetCookieHeader({
```

Comment:
Suggestion: Break this out into a different file.

To make the project easy to navigate we should only have a single header in each file.

Shared classes like `Cookie` can then be placed in some shared place.